Milestone 3: Game logic

ad9311 opened the pull request:
### Milestone 3: Game logic

In this PR we present (hopefully) a fully playable game.</br> The two players must only input letters for their names and numbers from 1-9.</br>  Whoever strikes a line either vertically, horizontally or diagonally first, wins.

Commit:
Add player class, loop and methods

Commit:
Fix win condition

Commit:
Add tie position

Commit:
Fix check_number method

ElijahTobs reviewed (changes requested):
# Status: Changes Required :recycle: 

## Hi @ad9311 and @AbdumurodovaZulfizar :wave: 

## HIGHLIGHTS 💯 
- The application does not allow a player to move into an occupied position ✔️ 
- The application does not exit/stop due to invalid input - it informs the player about it and asks for their move again ✔️ 

## REQUIRED ♻️ 

* [x] Your project should use at least two classes (Usually it uses three: Game, Board and Player). The code of the project should be logically divided between classes. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [x] All game-logic related resources (classes and/or modules) are to be kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [x] The application is expected to save players move and displays status of the board after that. Your application didn't display the status of the board when the winning move was made
![image](https://user-images.githubusercontent.com/16058470/118052015-ea678680-b379-11eb-9ab7-d50a5a34e3f7.png)
* [x] I tried to continue without a player name and I got the error below.
![image](https://user-images.githubusercontent.com/16058470/118052551-db350880-b37a-11eb-93a2-b7ad86eb98e2.png)
The player should be informed that they're required to enter their name.
* [x] You're expected to use `attr_readers`/`attr_writers`. You should only uses `attr_accessors` if you really need it and be aware that by using it, you allow writing that property from outside the class. That variable can be set to anything. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))



## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

Commit:
Add more classes

Remove .gitkeep

Commit:
Add permissions to be excecutable

Commit:
Add some emogies for strings

Commit:
Correct linter error and change condition

ad9311 commented:
We have corrected the following issues:
- Added two extra classes, board and game
- If player inputs an empty string, the game catches the error
- All logic resources are in lib/ only methods that contain `puts` and `gets.chomp` are in main.rb
- most atributes are only readable.
- The game outputs the last move for every player

mimipeshy commented:
## STATUS: APPROVED 🥳 👯 

Hello, team @ad9311 and @AbdumurodovaZulfizar 👋🏾 👋🏾 congratulations on making it this far in this milestone.
#### Highlights

-  You have worked on the TSE changes ✔️ 
- Your game works very well, keep it up 😄 ✔️ 

Therefore I approve of this project go ahead and merge. All the best in the next milestones. 

Regards
Peris Ndanu
